### DISCUSSION TITLE
Multiple per X songs Schedule

### DISCUSSION THREAD

**Discussion by zachryit** [savonet/liquidsoap#2729] [Q&A]
I have the following setup to play from jingle 1 (j1) after every 2 music tracks and play from jingle 2 (j2) after every 4 music tracks

```
music = playlist("music.m3u")
j1 = playlist("j1.m3u")
j2 = playlist("j2.m3u")

# 2 x tracks Playlists
radio = rotate( weights=[1, 2], [j1, music])

# 4 x tracks Playlists
radio = rotate( weights=[1, 4], [j2, radio])

output.dummy(radio, fallible=true)
```

logs
j2,j1,music,music,j1,j2,music,music,j1,music,j2...

From the logs, j1 plays after every 2 tracks and j2 plays after every 4 tracks which is expected.
But the requirement is for j2 to play after every 4 music tracks (excluding j1 in the count)

Please how do I make sure that j2 only plays after playing 4 music tracks (and not 2 j1 tracks and 2 music tracks)

**Comment by toots** [comment]
Hi,

When getting into complex scheduling like this, I would suggest to implement a custom scheduling. It is a function that returns the next song. It could implemented in liquidsoap scripting language or using any tool of your choosing via `process.lines` for instance.

The function is then used in `request.dynamic` to create a source playing the requests returned by the functions one after the other.

A custom function will offer much more flexibility than tweaking the complex rules of `rotate` and the like.
